chef-utils/lib/chef-utils/dsl/platform.rb

```diff
@@ -0,0 +1,339 @@
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def arch_platform?(node = __getnode)
+        node["platform"] == "arch"
+      end
+
+      # Determine if the current node is solaris2
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def solaris2_platform?(node = __getnode)
+        node["platform"] == "solaris2"
+      end
+
+      # Determine if the current node is smartos
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def smartos_platform?(node = __getnode)
+        node["platform"] == "smartos"
+      end
+
+      # Determine if the current node is omnios
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def omnios?(node = __getnode)
+        node["platform"] == "omnios"
+      end
+      alias_method :omnios_platform?, :omnios?
+
+      # Determine if the current node is openindiana
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def openindiana?(node = __getnode)
+        node["platform"] == "openindiana"
+      end
+      alias_method :openindiana_platform?, :openindiana?
+
+      # Determine if the current node is nexentacore
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def nexentacore?(node = __getnode)
+        node["platform"] == "nexentacore"
+      end
+      alias_method :nexentacore_platform?, :nexentacore?
+
+      # Determine if the current node is aix
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def aix_platform?(node = __getnode)
+        node["platform"] == "aix"
+      end
+
+      # Determine if the current node is freebsd
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def freebsd_platform?(node = __getnode)
+        node["platform"] == "freebsd"
+      end
+
+      # Determine if the current node is openbsd
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def openbsd_platform?(node = __getnode)
+        node["platform"] == "openbsd"
+      end
+
+      # Determine if the current node is netbsd
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def netbsd_platform?(node = __getnode)
+        node["platform"] == "netbsd"
+      end
+
+      # Determine if the current node is dragonflybsd
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def dragonfly_platform?(node = __getnode)
+        node["platform"] == "dragonfly"
+      end
+
+      # Determine if the current node is MacOS.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def mac_os_x_platform?(node = __getnode)
+        node["platform"] == "mac_os_x"
+      end
+      alias_method :macos_platform?, :mac_os_x_platform?
```

Comment:
can we make the real method macos_platform? and the alias mac_os_x_platform? since that's the one that needs to go away